src/main/java/mate/academy/springboot/datajpa/controller/CategoryController.java

```diff
@@ -0,0 +1,57 @@
+package mate.academy.springboot.datajpa.controller;
+
+import mate.academy.springboot.datajpa.dto.mapper.CategoryMapper;
+import mate.academy.springboot.datajpa.dto.request.CategoryRequestDto;
+import mate.academy.springboot.datajpa.dto.response.CategoryResponseDto;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final CategoryMapper categoryMapper;
+
+    public CategoryController(CategoryService categoryService,
+                              CategoryMapper categoryMapper) {
+        this.categoryService = categoryService;
+        this.categoryMapper = categoryMapper;
+    }
+
+    @PostMapping
+    public CategoryResponseDto saveCategory(@RequestBody
+                                                CategoryRequestDto categoryRequestDto) {
+        Category category = categoryService
+                .save(categoryMapper.toModel(categoryRequestDto));
+        return categoryMapper.toDto(category);
+    }
+
+    @GetMapping("/{id}")
+    public CategoryResponseDto getCategoryById(@PathVariable Long id) {
+        return categoryMapper.toDto(categoryService.getById(id));
+    }
+
+    @PutMapping("/{id}")
+    public CategoryResponseDto updateCategory(@PathVariable Long id,
+                                              @RequestBody CategoryRequestDto categoryRequestDto) {
+        Category category = categoryMapper.toModel(categoryRequestDto);
+        category.setId(id);
+        Category updatedCategory = categoryService.update(category);
+        return categoryMapper.toDto(updatedCategory);
+    }
+
+    @DeleteMapping("/{id}")
+    public String deleteCategory(@PathVariable Long id) {
+        categoryService.deleteById(id);
+        return String.format("Category with id={%s} was successfully deleted", id);
```

Comment:
In real cases, it’s not necessary. we operate response statuses to understand operation results.